armory/data/datasets.py

```diff
@@ -212,58 +206,73 @@ def _generator_from_tfds(
             dataset_dir, dataset_name=dataset_name,
         )
 
-    default_graph = tf.compat.v1.keras.backend.get_session().graph
+    if framework == "pytorch":
+        from armory.data.pytorch_loaders import ImageTFRecordDataSet
 
-    ds, ds_info = tfds.load(
-        dataset_name,
-        split=split_type,
-        as_supervised=as_supervised,
-        data_dir=dataset_dir,
-        with_info=True,
-        download_and_prepare_kwargs=download_and_prepare_kwargs,
-        shuffle_files=shuffle_files,
-    )
-    if not as_supervised:
-        try:
-            x_key, y_key = supervised_xy_keys
-        except (TypeError, ValueError):
-            raise ValueError(
-                f"When as_supervised=False, supervised_xy_keys must be a (x_key, y_key)"
-                f" tuple, not {supervised_xy_keys}"
-            )
-        if not isinstance(x_key, str) or not isinstance(y_key, str):
-            raise ValueError(
-                f"supervised_xy_keys be a tuple of strings,"
-                f" not {type(x_key), type(y_key)}"
-            )
-        ds = ds.map(lambda x: (x[x_key], x[y_key]))
-    if lambda_map is not None:
-        ds = ds.map(lambda_map)
-
-    ds = ds.repeat(epochs)
-    if shuffle_files:
-        ds = ds.shuffle(batch_size * 10, reshuffle_each_iteration=True)
-    if variable_length and batch_size > 1:
-        ds = ds.batch(1, drop_remainder=False)
+        if not shuffle_files:
+            raise ValueError("PyTorch DataLoaders from Armory are shuffled by default")
+        ds_name, ds_version = dataset_name.split(":")
```

Comment:
We should probably document that this shuffling is due to dareblopy, not due to the defaults of PyTorch iter dataloaders.